flink-clients/src/main/java/org/apache/flink/client/program/StreamContextEnvironment.java

```diff
@@ -161,6 +213,18 @@ public static void setAsContext(
             final boolean suppressSysout) {
         StreamExecutionEnvironmentFactory factory =
                 conf -> {
+                    final List<String> errors = new ArrayList<>();
+                    final boolean allowConfigurations =
+                            configuration.getBoolean(
+                                    DeploymentOptions.ALLOW_CLIENT_JOB_CONFIGURATIONS);
+                    if (!allowConfigurations && !conf.toMap().isEmpty()) {
+                        conf.toMap()
+                                .forEach(
+                                        (k, v) ->
+                                                errors.add(
+                                                        ConfigurationNotAllowedMessage
+                                                                .ofConfigurationKeyAndValue(k, v)));
```

Comment:
Would throwing an exception here right away make more sense? It could point user to the exact location where the problem is.

Also it would avoid passing the errors into the stream environment, which could simplify the change-set a bit.

eg.

```suggestion
                        throw new MutatedConfigurationException("Supplying a custom configuration for the stream environment is not allowed, because the client-side configuration is disabled.");
```

Reply:
Actually, I like that all errors are collected and users immediately see what they are not allowed to set. Otherwise, it might take multiple submissions until they have seen all errors. Regarding the code complexity, it seems okay to me since we are only adding to the `StreamContextEnvironment`. In fact, I am a bit surprised that the `StreamContextEnvironment` is marked as `@PublicEvolving` not sure when users will ever interact with it.